packages/cli/src/runners/task-broker.service.ts

```diff
@@ -75,15 +75,11 @@ export class TaskBroker {
 
 	expireTasks() {
 		const now = process.hrtime.bigint();
-		const invalidOffers: number[] = [];
-		for (let i = 0; i < this.pendingTaskOffers.length; i++) {
-			if (this.pendingTaskOffers[i].validUntil < now) {
-				invalidOffers.push(i);
+		for (let i = this.pendingTaskOffers.length; i >= 0; i--) {
+			if (this.pendingTaskOffers[i]?.validUntil < now) {
+				this.pendingTaskOffers.splice(i, 1);
```

Comment:
I'd expect to reverse loop like this:

```suggestion
		for (let i = this.pendingTaskOffers.length - 1; i >= 0; i--) {
			if (this.pendingTaskOffers[i].validUntil < now) {
				this.pendingTaskOffers.splice(i, 1);
```